.github/workflows/rust-build-test.yaml

```diff
@@ -0,0 +1,26 @@
+name: Rust
+
+on:
+    push:
+        branches: ["main"]
+    pull_request:
+        branches: ["main"]
+
+env:
+    CARGO_TERM_COLOR: always
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Build
+              run: cargo build --release --verbose
+
+            - name: Run tests
+              run: cargo test --verbose
```

Comment:
Can you run `make tests` here? Note that that will also run the linter and fail if there is a linting problem.

If you want you can add the `--verbose` flag to the Makefile